qiskit/algorithms/minimum_eigen_solvers/adapt_qaoa.py

```diff
@@ -0,0 +1,539 @@
+    num_qubits: int, add_single: bool = True, add_multi: bool = True, pool_type: str = None
+) -> List:
+    """
+    Gets all combinations of mixers in desired set (standard qaoa mixer, single qubit
+        mixers, multi qubit mixers)
+    Args:
+        num_qubits: number of qubits
+        add_single: whether to add single qubit to mixer pool (not standard qaoa x mixers)
+        add_multi: whether to add multi qubit to mixer pool
+        pool_type: Optional input overrides add_single and add_multi by respecifying
+            these conditions based on the preset mixer pool classes: 'multi',
+            'singular' and 'single'.
+
+    Returns:
+        List of all possible combinations of mixers.
+
+    Raises:
+        ValueError: If an unrecognisible mixer type has been provided.
+    """
+    if pool_type:
+        if pool_type == "multi":
+            add_multi, add_single = True, True
+        elif pool_type == "singular":
+            add_multi, add_single = False, True
+        elif pool_type == "single":
+            add_multi, add_single = False, False
+        else:
+            raise ValueError(
+                "Unrecognised mixer pool type {}, modify this input to the available presets"
+                " 'single', 'singular' or 'multi'."
+            )
+
+    # always include the all x's:
+    mixer_pool = ["X" * num_qubits]
+    if add_single:
+        # y's
+        mixer_pool.append("Y" * num_qubits)
+        mixer_pool += [i * "I" + "X" + (num_qubits - i - 1) * "I" for i in range(num_qubits)]
+        mixer_pool += [i * "I" + "Y" + (num_qubits - i - 1) * "I" for i in range(num_qubits)]
+    if add_multi:
+        indicies = list(permutations(range(num_qubits), 2))
+        indicies = list(set(tuple(sorted(x)) for x in indicies))
+        combos = list(combinations_with_replacement(["X", "Y", "Z"], 2))
+        full_multi = list(product(indicies, combos))
+        for item in full_multi:
+            iden_str = list("I" * num_qubits)
+            iden_str[item[0][0]] = item[1][0]
+            iden_str[item[0][1]] = item[1][1]
+            mixer_pool.append("".join(iden_str))
+
+    op_dict = {"I": I, "X": X, "Y": Y, "Z": Z}
+
+    def is_all_same(items):
+        return all(x == items[0] for x in items)
+
+    def string_to_op(qstring):
+        if is_all_same(qstring):
+            # case where its all X's or Y's
+            gate = qstring[0]
+            list_string = [
+                i * "I" + gate + (len(qstring) - i - 1) * "I" for i in range(len(qstring))
+            ]
+            return sum(
+                [
+                    reduce(lambda a, b: a ^ b, [op_dict[char.upper()] for char in x])
+                    for x in list_string
+                ]
+            )
+        return reduce(lambda a, b: a ^ b, [op_dict[char.upper()] for char in qstring])
+
+    mixer_pool = [string_to_op(mixer) for mixer in mixer_pool]
+    return mixer_pool
+
+
+# TODO: Fix the following issues for unittest:
```

Comment:
Is this still open?

Reply:
Yes, some of the unittests still do not pass. The unittests are based on those in the QAOA unittests so the tests should mostly have the same outcomes (I have modified the ones that are expected to have a different output) but not all seem to be passing at the moment.